projects/103_homework/src/main.c

```diff
@@ -0,0 +1,60 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// interrupt call back function
+void interrupt_callback(const GpioAddress *address, void *context) {
+  GpioAddress *ADC_address = context;
+
+  uint16_t ADC_reading = 0;
```

Comment:
nit: We generally don't use capitals in variable names. `adc_reading` would be much more compliant.